meilisearch-http/src/lib.rs

```diff
@@ -145,18 +146,52 @@ pub fn dashboard(config: &mut web::ServiceConfig, _enable_frontend: bool) {
 macro_rules! create_app {
     ($data:expr, $auth:expr, $enable_frontend:expr, $opt:expr, $analytics:expr) => {{
         use actix_cors::Cors;
+        use actix_web::dev::Service;
         use actix_web::middleware::TrailingSlash;
         use actix_web::App;
         use actix_web::{middleware, web};
         use meilisearch_http::error::MeilisearchHttpError;
+        use meilisearch_http::metrics;
         use meilisearch_http::routes;
         use meilisearch_http::{configure_data, dashboard};
         use meilisearch_types::error::ResponseError;
 
+        use lazy_static::lazy_static;
+        use prometheus::{
+            opts, register_histogram_vec, register_int_counter_vec, register_int_gauge,
+        };
+        use prometheus::{HistogramTimer, HistogramVec, IntCounterVec, IntGauge};
+
         App::new()
             .configure(|s| configure_data(s, $data.clone(), $auth.clone(), &$opt, $analytics))
             .configure(routes::configure)
             .configure(|s| dashboard(s, $enable_frontend))
+            .wrap_fn(|req, srv| {
+                let mut histogram_timer: Option<HistogramTimer> = None;
+                let request_path = req.path();
+                let is_registered_resource = req.resource_map().has_resource(request_path);
+                if is_registered_resource {
+                    let request_method = req.method().to_string();
+                    histogram_timer = Some(
+                        metrics::HTTP_RESPONSE_TIME_SECONDS
+                            .with_label_values(&[&request_method, request_path])
+                            .start_timer(),
+                    );
+                    metrics::HTTP_REQUESTS_TOTAL
+                        .with_label_values(&[&request_method, request_path])
+                        .inc();
+                }
```

Comment:
From what I see/understand, Prometheus is using [`Atomic`](https://docs.rs/prometheus/latest/src/prometheus/atomic64.rs.html) to update its state.
Most access are relaxed, but the `inc` is an `Acquire` for example (which can be slower than a mutex iirc).

And this code is executed for EVERY call to ANY route.

Maybe it could slow down mielisearch when it’s under high pressure?
What do you think @meilisearch/core-team?

Personally, I still think we should accept this PR, my question is, should it work like that, or should we put the feature behind a cli args so it only slows down those who need prometheus?

Reply:
Good question @irevoire, I agree with you that this PR is a great improvement.

I like the idea of setting an opt-in flag for this feature as we made for the auto-batching beta feature.
Moreover, this feature didn't pass the usual feature process and didn't have any approval from the product team.

So, I'd find it safer to deactivate by default this feature. 

But again, it's a great feature that could be useful for some of our users and for us. 😊

Reply:
This PR looks great and would be incredibly helpful for us on the cloud side.

Having this be an opt-in feature would be fine (we would likely enable it by default for all projects)

Reply:
@ManyTheFish It seems it could be a strong basis for a more [detailed exporter](https://github.com/meilisearch/product/issues/496) that is considered for the future.